### PR TITLE
[KernelGen][Nvidia] Add rnn_relu_cell operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `rnn_relu_cell`. Applies a single-layer Elman RNN cell with ReLU activation:
out = relu(input @ w_ih^T + hx @ w_hh^T + b_ih + b_hh).

## Testing
- Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (9 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia

## Repository / Branch
- Target repository: `flagos-ai/FlagGems`
- Base branch: `master`

## Performance
Test command: `pytest benchmark/test_rnn_relu_cell.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 16, 128 | 0.015 | 0.300 | 0.051 |
| 64, 512 | 0.018 | 0.311 | 0.057 |
| 256, 1024 | 0.026 | 0.317 | 0.081 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 16, 128 | 0.022 | 0.377 | 0.057 |
| 64, 512 | 0.028 | 0.359 | 0.078 |
| 256, 1024 | 0.080 | 0.479 | 0.167 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| 16, 128 | 0.015 | 0.301 | 0.051 |
| 64, 512 | 0.018 | 0.304 | 0.058 |
| 256, 1024 | 0.027 | 0.299 | 0.089 |

**Arithmetic Mean Speedup: 0.077x**
## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (9 cases, --level core) | 0.077 | Primary |
| Tianshu | — | — | — | — |
| Muxi | — | — | — | — |
| Ascend | — | — | — | — |
| Hygon | — | — | — | — |

## Files Changed
- `src/flag_gems/ops/rnn_relu_cell.py`: Triton kernel implementation
- `tests/test_rnn_relu_cell.py`: Accuracy test
- `benchmark/test_rnn_relu_cell.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

### CODE REVIEW

**Comment by bin913** [comment]
> ## Summary
> Adds a Triton kernel for `rnn_relu_cell`. Applies a single-layer Elman RNN cell with ReLU activation: out = relu(input @ w_ih^T + hx @ w_hh^T + b_ih + b_hh).
> 
> ## Testing
> * Parametrized tests over multiple shapes and float16/float32/bfloat16 dtypes (9 cases)
> * Validated against PyTorch reference on device side
> * Tested on: Nvidia
> 
> ## Repository / Branch
> * Target repository: `flagos-ai/FlagGems`
> * Base branch: `master`
> 
> ## Performance
> Test command: `pytest benchmark/test_rnn_relu_cell.py --level core` (NVIDIA H20)
> 
> ### float16
> Size Detail	Torch Latency (ms)	Gems Latency (ms)	Speedup
> 16, 128	0.015	0.300	0.051
> 64, 512	0.018	0.311	0.057
> 256, 1024	0.026	0.317	0.081
> ### float32
> Size Detail	Torch Latency (ms)	Gems Latency (ms)	Speedup
> 16, 128	0.022	0.377	0.057
> 64, 512	0.028	0.359	0.078
> 256, 1024	0.080	0.479	0.167
> ### bfloat16
> Size Detail	Torch Latency (ms)	Gems Latency (ms)	Speedup
> 16, 128	0.015	0.301	0.051
> 64, 512	0.018	0.304	0.058
> 256, 1024	0.027	0.299	0.089
> **Arithmetic Mean Speedup: 0.077x**
> 
> ## Multi-backend Testing
> Backend	Accuracy Test	Benchmark	Speedup (mean)	Notes
> Nvidia (H20)	PASS	PASS (9 cases, --level core)	0.077	Primary
> Tianshu	—	—	—	—
> Muxi	—	—	—	—
> Ascend	—	—	—	—
> Hygon	—	—	—	—
> ## Files Changed
> * `src/flag_gems/ops/rnn_relu_cell.py`: Triton kernel implementation
> * `tests/test_rnn_relu_cell.py`: Accuracy test
> * `benchmark/test_rnn_relu_cell.py`: Performance benchmark
> * `src/flag_gems/ops/__init__.py`: Register import and `__all__`
> * `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
> * `conf/operators.yaml`: Add operator entry (kind: NeuralNetwork, stage: alpha 5.4)

Why is the performance speedup ratio so low? Can we optimize it further to get better performance?

File: conf/operators.yaml [hunk 1]
```diff
@@ -12917,6 +12917,20 @@ ops:
...
+  for:
+  - rnn_relu_cell
+  labels:
+  - aten
+  - KernelGen
+  - NoCPU
+  kind:
+  - NeuralNetwork
+  stages:
+  - alpha: '5.4'
```
Comment (bin913) [inline conf/operators.yaml]:
```suggestion
  - alpha: '5.5.0'
```

File: tests/test_rnn_relu_cell.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+@pytest.mark.skipif(
+    cfg.TO_CPU or flag_gems.device != "cuda" or not torch.cuda.is_available(),
+    reason="Triton kernel is CUDA-only",
+)
+@pytest.mark.rnn_relu_cell
+def test_rnn_relu_cell_smoke_cell_module():
+    """Smoke test: one step of torch.nn.RNNCell (ReLU) equals rnn_relu_cell."""
+    batch, input_size, hidden_size = 4, 8, 8
+    dtype = torch.float32
```
Comment (bin913) [inline tests/test_rnn_relu_cell.py:125]: Why do we only test fp32 here?
